src/redux/balanceSheets/balanceSheetsSlice.js

```diff
@@ -0,0 +1,47 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const balanceSheetsURL = 'https://financialmodelingprep.com/api/v3/balance-sheet-statement/';
+
+const initialState = {
+  balanceSheets: [],
+  isLoading: false,
+  error: undefined,
+};
+
+export const getBalanceSheetItems = createAsyncThunk(
+  'balanceSheets/getBalanceSheetItems',
+  async (symbol, thunkAPI) => {
+    try {
+      const resp = await axios(`${balanceSheetsURL}${symbol.symbol}?apikey=b0f32ea981cc58fafa1177563647a780&limit=120`);
+      return resp.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue(error);
```

Comment:
- [ ] Based on project requirements, you meant to use a Filter stateless component to filter data retrieved from the API,  However, you didn't include the Filter stateless component. The code is focused on fetching `balancesheet` data using Redux and rendering it. There is no filtering functionality implemented in this component.

- [ ] To add filtering functionality, you need to create a separate component responsible for filtering the `balanceSheets` data. This new component should receive the `balanceSheets` data as props, apply the filtering logic, and render the filtered data.

### Sample code as a guide to you:
```
import React, { useState } from 'react';
import { useSelector } from 'react-redux';

const BalanceSheetFilter = () => {
  const balanceSheets = useSelector((state) => state.balanceSheets.balanceSheets);
  const [filteredBalanceSheets, setFilteredBalanceSheets] = useState([]);

  const handleFilter = (filterValue) => {
    const filteredData = balanceSheets.filter((balanceSheet) =>
      balanceSheet.someProperty === filterValue
    );
    setFilteredBalanceSheets(filteredData);
  };

  return (
    <div>
      {/* Render filter options and input */}
      {/* Add filter logic and update filteredBalanceSheets state */}
      {/* Render the filtered data */}
    </div>
  );
};

export default BalanceSheetFilter;
``